src/ui/backend/BaseBacking.js

```diff
@@ -62,36 +83,39 @@ export default class BaseBacking {
 
   copy () {
     var copy = {};
+    for (var key in BASE_STYLE_PROPS) {
+      copy[key] = this[key];
+    }
 
-    for (var i = 0; i < styleKeyList.length; i++) {
-      var key = styleKeyList[i];
+    for (var i = 0; i < extendedStylePropList.length; i++) {
+      var key = extendedStylePropList[i];
       copy[key] = this[key];
     }
 
     return copy;
   }
 
   update (style) {
-    for (var i = 0; i < styleKeyList.length; i++) {
-      var key = styleKeyList[i];
+    for (var key in style) {
+      if (this[key] !== undefined) {
+        this[key] = style[key];
+      }
+    }
+
+    for (var i = 0; i < extendedStylePropList.length; i++) {
+      var key = extendedStylePropList[i];
       if (style[key] !== void 0) {
         this[key] = style[key];
       }
     }
+
     return this;
   }
 
 };
 
-var styleKeys = BaseBacking.prototype.constructor.styleKeys = {};
-var styleKeyList = [];
-
+var extendedStylePropList = [];
```

Comment:
The problem with the boolean is that I am afraid that all the methods that accept views won't be optimized (or will be deoptimized) because the hidden classes of the treated views will be different (some of them will have layout properties and some wont). I could give it a try anyway.

In my opinion, the ideal would be to have 2 different classes but that would imply some refactoring.